src/WebJobs.Script/Diagnostics/ILoggerFactoryHook.cs

```diff
@@ -0,0 +1,21 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Azure.WebJobs.Script.Diagnostics
+{
+    /// <summary>
+    /// Provides ways to plug into the ScriptHost ILoggerFactory initialization.
+    /// </summary>
+    public interface ILoggerFactoryHook
```

Comment:
Let me do a minor refactoring here. I can make a default Builder and allow you to replace it (or derive from it, which the CLI can do). I have dreams of allowing people to plug in their own ILoggerProviders someday, so this gets us closer to that.